File: apps/android/app/src/main/java/za/co/proplyst/app/data/leases/Lease.kt
package za.co.proplyst.app.data.leases

// Domain model -- mirrors packages/types/src/leasing.ts's Lease interface, minus source/
// sourceDocumentId/sourceApplicationId (internal provenance fields with no view-only-screen use
// yet, same reasoning as Tenant.idNumberRef being left out of the Tenants slice).
data class Lease(
    val id: String,
    val orgId: String,
    val unitId: String,
    val startDate: String,
    val endDate: String?,
    val rentAmount: Double,
    val rentFrequency: String,
    val depositAmount: Double,
    val status: String,
)

File: apps/ios/Sources/Proplyst/Domain/Lease.swift
import Foundation

/// Mirrors `packages/types/src/leasing.ts`'s `Lease` interface and `apps/android`'s `Lease`
/// domain model. `source`/`sourceDocumentId`/`sourceApplicationId` deliberately omitted --
/// internal provenance fields with no view-only-screen use yet, same reasoning Android's own
/// model documents.
struct Lease: Identifiable, Codable, Equatable, Sendable {
    let id: String
    let orgId: String
    let unitId: String
    let startDate: String
    let endDate: String?
    let rentAmount: Double
    let rentFrequency: String
    let depositAmount: Double
    let status: String

    enum CodingKeys: String, CodingKey {
        case id
        case orgId = "org_id"
        case unitId = "unit_id"
        case startDate = "start_date"
        case endDate = "end_date"
        case rentAmount = "rent_amount"
        case rentFrequency = "rent_frequency"
        case depositAmount = "deposit_amount"
        case status
    }
}
